### ISSUE TITLE
Focal loss implementation

### ISSUE THREAD

**Issue by b02202050** [pytorch/pytorch#35882]
## 🚀 Feature

Implementation of Focal loss, which is first introduced in [Tsung-Yi Lin, et al. "Focal Loss for Dense Object Detection," 2018.](https://arxiv.org/abs/1708.02002).

## Motivation
Focal loss is a powerful type of loss to alleviate class imbalance problem, which is a commonly seen situation in most of the learning tasks.

## Pitch
implement focal loss for both multi-class(softmax) and multi-label(sigmoid) version.

## Additional context

cc @fmassa

**Comment by jerryzh168** [comment]
I feel that should be in the domain specific library, e.g. torchvision. cc @fmassa

**Comment by alaydshah** [comment]
I would like to implement this loss function

**Comment by fmassa** [comment]
Hi,

Focal loss is available in torchvision since the 0.8.0 release, see the release notes in https://github.com/pytorch/vision/releases/tag/v0.8.0 and implementation in https://github.com/pytorch/vision/blob/v0.8.0/torchvision/ops/focal_loss.py

Given that focal loss is for now specific to computer vision, I would say that this should stay in torchvision, and as we have already implemented it there, I'm closing this issue, but let us know if you think differently/

**Comment by ZhiyuanChen** [comment]
> I would like to implement this loss function

I disagree. Focal Loss is very useful for imbalanced dataset.
